### PR TITLE
Backup settings screen

### LINKED ISSUE #149: implement Settings -> Backups screen
follow behavior of Android app in situations when there is no backup present, or when there is

### PR DESCRIPTION
**Pull request by Kharchevskyi**
close #149

### CODE REVIEW

**Comment by Kharchevskyi** [comment]
Simulator Screen Shot - iPhone Xʀ - 2020-09-27 at 14 54 41
Simulator Screen Shot - iPhone Xʀ - 2020-09-27 at 14 55 31
Simulator Screen Shot - iPhone Xʀ - 2020-09-27 at 14 58 15

**Comment by Kharchevskyi** [comment]
Simulator Screen Shot - iPhone Xʀ - 2020-10-01 at 15 59 14
Simulator Screen Shot - iPhone Xʀ - 2020-10-01 at 15 59 19

**Comment by Kharchevskyi** [comment]
Simulator Screen Shot - iPhone 11 Pro - 2020-10-17 at 22 59 03

File: FlowCrypt/Controllers/Compose/ComposeViewController.swift [hunk 1]
```diff
@@ -328,7 +328,7 @@ extension ComposeViewController {
         to: [String],
         cc: [String] = [],
         bcc: [String] = [],
-        atts: [SendableMsg.Att] = []
+        atts: [SendableMsg.Attribute] = []
```
Comment (tomholub) [inline FlowCrypt/Controllers/Compose/ComposeViewController.swift:331]: These are `Att` as for attachment. If it's confusing then call them `Attachment`.

File: FlowCrypt/Controllers/Settings/Backup/Backups Scene/BackupViewController.swift [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    private enum Parts: Int, CaseIterable {
+        case info, action
+    }
+
+    enum State {
+        case idle
+        case backups([KeyDetails])
+        case noBackups
+
+        var isAnyBackups: Bool {
```
Comment (tomholub) [inline FlowCrypt/Controllers/Settings/Backup/Backups Scene/BackupViewController.swift:22]: `hasAnyBackups` would be more correct

File: FlowCrypt/Functionality/Services/BackupService/BackupService.swift [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            }
+
+            let privateKeyContext = keys
+                .compactMap { $0 }
+                .joinedPrivateKey
+
+            let privateKeyData = privateKeyContext.data().base64EncodedString()
+
+            let filename = "flowcrypt-backup-\(userId.email.userReadableEmail).key"
+            let messageAttributes = [SendableMsg.Attribute(name: filename, type: "text/plain", base64: privateKeyData)]
```
Comment (tomholub) [inline FlowCrypt/Functionality/Services/BackupService/BackupService.swift:85]: attachments

Comment (tomholub) [reply FlowCrypt/Functionality/Services/BackupService/BackupService.swift:85]: You can see the surrounding code - it's obvious it deals with files

File: FlowCrypt/Core/CoreTypes.swift [hunk 4]
```diff
@@ -125,7 +119,7 @@ struct SendableMsg {
     let from: String
     let subject: String
     let replyToMimeMsg: String?
-    let atts: [Att]
+    let atts: [Attribute]
```
Comment (tomholub) [inline FlowCrypt/Core/CoreTypes.swift:122]: Attachment

**Review by tomholub — CHANGES_REQUESTED** [review]
Looks good overall but please be mindful of purpose of names when renaming. See comments

File: FlowCrypt/Controllers/Compose/ComposeViewController.swift (hunk 1 above)
Comment (Kharchevskyi) [reply FlowCrypt/Controllers/Compose/ComposeViewController.swift:331]: Will fix this, thank you

**Review by tomholub — APPROVED** [review]

**Pull request merged**